apps/studio/src/features/editing-experience/components/LinkEditor/LinkHrefEditor.tsx

```diff
@@ -33,21 +42,40 @@ export const LinkHrefEditor = ({
   description,
   isRequired,
   isInvalid,
+  errorMessage,
+  setErrorMessage,
+  clearErrorMessage,
   pageLinkElement,
   fileLinkElement,
 }: LinkHrefEditorProps) => {
   const linkType = getLinkHrefType(value)
   const [selectedLinkType, setSelectedLinkType] = useState(linkType)
+  const [linkValueHistory, setLinkValueHistory] = useState({
+    ...INITIAL_LINK_VALUE_HISTORY,
+    ...{
+      [linkType]: value,
+    },
+  } as LinkValueHistory)
 
-  const handleLinkTypeChange = (value: string) => {
-    setSelectedLinkType(value)
-    onChange("")
-  }
+  // hacky solution to allow setting the correct state of "Add link" CTA button
+  // while not showing error message on initial render for better UX
+  const [isLinkTypeInitialRender, setIsLinkTypeInitialRender] = useState(true)
+  const shouldShowErrorState =
+    !isLinkTypeInitialRender &&
+    errorMessage !== "" &&
+    errorMessage !== undefined
 
   const { getRootProps, getRadioProps } = useRadioGroup({
     name: "link-type",
     defaultValue: linkType,
-    onChange: handleLinkTypeChange,
+    onChange: (newLinkType) => {
+      setIsLinkTypeInitialRender(true)
+      setSelectedLinkType(newLinkType)
```

Comment:
this might have been fixed by the other pr when i removed the switch case but i can't confirm